constraint/pkg/client/drivers/rego/driver.go

```diff
@@ -297,25 +310,58 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 
 		kindResults, err := drivers.ToResults(constraintsMap, resultSet)
 		if err != nil {
-			return nil, nil, err
-		}
-
-		for _, result := range kindResults {
-			result.EvaluationMeta = EvaluationMeta{
-				TemplateRunTime: float64(evalEndTime.Nanoseconds()) / 1000000,
-				ConstraintCount: uint(len(kindResults)),
-			}
+			return nil, err
 		}
 
 		results = append(results, kindResults...)
+
+		if d.gatherStats || (driverCfg != nil && driverCfg.StatsEnabled) {
+			statsEntries = append(statsEntries,
+				&instrumentation.StatsEntry{
+					Scope:    instrumentation.TemplateScope,
+					StatsFor: kind,
+					Stats: []*instrumentation.Stat{
+						{
+							Name:  templateRunTimeNanosName,
+							Value: uint64(evalEndTime.Nanoseconds()),
+							Source: instrumentation.Source{
+								Type:  instrumentation.EngineSourceType,
+								Value: instrumentation.RegoEngineSource,
+							},
+						},
+						{
+							Name:  constraintCountName,
+							Value: len(kindConstraints),
+							Source: instrumentation.Source{
+								Type:  instrumentation.EngineSourceType,
+								Value: instrumentation.RegoEngineSource,
+							},
+						},
+					},
+					Labels: []*instrumentation.Label{
+						{
+							Name:  "TracingEnabled",
+							Value: d.traceEnabled || (engineCfg != nil && engineCfg.TracingEnabled),
+						},
+						{
+							Name:  "PrintEnabled",
+							Value: d.printEnabled,
+						},
+						{
+							Name:  "EnableExternalDataClientAuth",
+							Value: d.enableExternalDataClientAuth,
```

Comment:
Let's omit this for now. It should be easy to infer by other means and should not impact performance.